sdk/cosmos/azure-cosmos-spark_3-1_2-12/Samples/Python/NYC-Taxi-Data/01_Ingestion.ipynb

```diff
@@ -0,0 +1,271 @@
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "inputWidgets": {},
+     "nuid": "2805de61-e652-4be6-a3e5-b084cd848dc9",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "** Sample - ingesting the NYC Green Taxi data into Cosmos DB**\n",
+    "\n",
+    "By setting the taregt throughput threshold to 0.8 (80%) we reduce throttling but still allow the ingestion to consume most of the provisioned throughput. For scenarios where ingestion shoudl only take a smaller subset of the available throughput this threshold can be reduced accordingly."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "inputWidgets": {},
+     "nuid": "56fc9432-df8b-456d-97c1-498c90938ed3",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "outputs": [],
+   "source": [
+    "writeCfg = {\n",
+    "  \"spark.cosmos.accountEndpoint\": cosmosEndpoint,\n",
+    "  \"spark.cosmos.accountKey\": cosmosMasterKey,\n",
+    "  \"spark.cosmos.database\": \"SampleDatabase\",\n",
+    "  \"spark.cosmos.container\": \"GreenTaxiRecords\",\n",
+    "  \"spark.cosmos.write.strategy\": \"ItemOverwrite\",\n",
+    "  \"spark.cosmos.write.maxConcurrency\": \"150\",\n",
+    "  \"spark.cosmos.write.bulkEnabled\": \"true\",\n",
+    "  \"spark.cosmos.throughputControlEnabled\": \"true\",\n",
+    "  \"spark.cosmos.throughputControl.name\": \"NYCGreenTaxiDataIngestion\",\n",
+    "  \"spark.cosmos.throughputControl.targetThroughputThreshold\": \"0.8\",\n",
+    "  \"spark.cosmos.throughputControl.globalControl.database\": \"SampleDatabase\",\n",
+    "  \"spark.cosmos.throughputControl.globalControl.container\": \"ThrouhgputControl\",\n",
+    "}\n",
+    "\n",
+    "df_NYCGreenTaxi_Input = spark.sql('SELECT * FROM source')\n",
+    "\n",
+    "df_NYCGreenTaxi_Input \\\n",
+    "  .write \\\n",
+    "  .format(\"cosmos.items\") \\\n",
+    "  .mode(\"Append\") \\\n",
+    "  .options(**writeCfg) \\\n",
+    "  .save()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "inputWidgets": {},
+     "nuid": "44c33e38-aa02-45f1-8e36-5c20a4fb6b35",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "**Cleanup - deleting the Cosmos DB container and database again (to reduce cost) - skip this step if you want to keep them**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "inputWidgets": {},
+     "nuid": "8afb7129-ae8c-4cc9-b177-58be5cc02445",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "outputs": [],
+   "source": [
+    "%sql\n",
+    "DROP TABLE IF EXISTS SampleDatabase.GreenTaxiRecords;\n",
+    "\n",
+    "DROP TABLE IF EXISTS SampleDatabase.ThrouhgputControl;\n",
+    "\n",
+    "DROP DATABASE IF EXISTS cosmosCatalog.SampleDatabase CASCADE;"
```

Comment:
`CASCADE` what does this do?
recursively drops the tables in the database?

Reply:
Yes